angular/projects/spark-angular/src/lib/components/sprk-alert/sprk-alert.component.ts

```diff
@@ -118,12 +152,49 @@ export class SprkAlertComponent {
     return alertClassArray.join(' ');
   }
 
+  /**
+   * @ignore
+   */
+  getIconName(): string {
+    const variantToUse = this.variant || this.alertType;
+    const variantIcons = {
+      success: 'check-mark-filled',
+      info: 'bell-filled',
+      fail: 'exclamation-filled',
+    };
+    const icon =
+      this.iconName !== undefined ? this.iconName : variantIcons[variantToUse];
+
+    return icon;
+  }
+
+  /**
+   * @ignore
+   */
+  // TODO: Remove when `dismissible` is deprecated - issue #1281.
+  getIsAlertDismissible(): boolean {
+    if (this.isDismissible === undefined && this.dismissible !== undefined) {
+      this.isAlertDismissible = this.dismissible;
+    } else if (this.isDismissible !== undefined) {
+      this.isAlertDismissible = this.isDismissible;
+    } else {
+      this.isAlertDismissible = true;
+    }
+    return this.isAlertDismissible;
+  }
+
   /**
    * When the dismiss button is clicked
-   * this method sets the visable state to `false`
+   * this method sets `isVisible` to `false`
    * and hides the Alert component.
    */
   alertDismiss(event): void {
-    this.visible = false;
+    this.isVisible = false;
+    this.dismissEvent.emit();
+  }
+
+  // TODO: Remove when `dismissible` is deprecated - issue #1281.
+  ngOnInit(): void {
+    this.getIsAlertDismissible();
```

Comment:
Calculating this in `ngOnInit` means it won't be recalculated if the value is updated after init. Check out `sprk-tooltip` or `sprk-dictionary` for examples on how I got around this.

```
it('should correctly respond to changes to dismissable', () => {
 component.dismissible = true;
 fixture.detectChanges();
 expect(fixture.nativeElement.querySelectorAll('button').length).toEqual(1);
 component.dismissible = false;
 fixture.detectChanges();
 expect(fixture.nativeElement.querySelectorAll('button').length).toEqual(0);
});
```